pkgs/data/icons/windows10-icons/default.nix

```
@@ -0,0 +1,25 @@
+{ lib, stdenv, fetchurl }:
+stdenv.mkDerivation rec {
+  pname = "windows10-icons";
+  version = "1.0";
+
+  src = fetchurl {
+    url = "https://github.com/B00merang-Artwork/Windows-10/archive/refs/tags/${version}.tar.gz";
+    sha256 = "sha256-ZGuLyOBNOANIicqs7zhGy2ZVmV8YBdh7P/e8Cykkmq4=";
+  };
+
+  installPhase = ''
+    install -Dm644 $(find . -type f -maxdepth 1) -t $out/share/icons/windows10
+    for dir in $(find . -type d -maxdepth 1)
+      do install -d $dir $out/share/icons/windows10/$dir
+    done
+  '';
+
+  meta = with lib; {
+    description = "Windows 10 icon theme";
+    homepage = "http://b00merang.weebly.com/windows-10.html";
+    license = licenses.gpl3;
```

```suggestion
    license = licenses.unfree;
```
These icons have been ripped out of Windows, they are certainly not licensed under the GPL.

---

Very fair, I was originally thinking the same (see comments on issue this closes). Are we still allowed to include it?

---

I'm not really sure.

There is lots of unfree software in nixpkgs. Marking software as unfree results in the CI not building it and hence it not being available in the NixOS binary cache. It will be transparently built on the users' machines.

For this reason I think including anything in nixpkgs should be legally fine as long as it's marked as unfree -- NixOS infrastructure won't touch it, so it essentially is the same as just linking to the project. I'm not a lawyer and this is not legal advice.

Whether we want to include things that are pretty much blatant copyright infringement in nixpkgs is another thing. I can't find a precedent for this at first glance, but there probably has been one.

---

Good point. I'm not really sure I want to be the one opening the flood gates and setting precedent either, but nonetheless I've committed the change - we'll see whether this gets merged or not I suppose. :)